### DISCUSSION TITLE
Is it possible to reference an attribute value with raw text while calling Contains()?

### DISCUSSION THREAD

**Discussion by christian-townsend** [datalust/seq-tickets#1339] [Help]
I would like to perform a search using Contains() by combining raw text with an Attribute value (id). See below:

`Select callid as id from stream 
where Contains(Message, 'UniqueId: {id}' )
limit 1000`

Is this possible in Seq?

**Comment by nblumhardt** [comment]
Hi @christian-townsend,

Yes, `concat` will do that:

```sql
select callid as id
from stream
where contains(@Message, concat('UniqueId: ', id))
limit 1000
```

(Assuming that by `Message` you mean `@Message`.)

If `id` is not a string, you can convert it to one using `tojson`, e.g. `concat('UniqueId: ', tojson(id))`.

Let me know if this helps!

Best regards,
Nick

**Reply by christian-townsend** [reply]
Thank you, Nick. This is great, and works well. One question, how would I do the inverse of this? 
If my goal is to find all callids where the value in `@Message` next to UniqueId is not the `callid`, would this be sufficient? 

```
select callid as id
from stream
where contains(@Message, concat('UniqueId: ', not id))
limit 1000
```

Basically, I want to find instances  of a `callid` where the value that comes after UniqueId is not equal to the `callid`.

**Reply by nblumhardt** [reply]
Hi Christian,

I think what you're after will be something like:

```
select callid as id
from stream
where not contains(@Message, concat('UniqueId: ', id))
limit 1000
```

Would be great to hear how you go!

**Reply by christian-townsend** [reply]
Hi Nick, 

This solution unfortunately omits the "UniqueId" portion of what I am still looking for. 
I still would like to return events that contain "UniqueId" in the `@Message` but the value that follows doesn't need to be equal to the `callid`

E.g. I have an event with a `callid` equal to 1234, and I also have a `@Message` that contains 'UniqueId: 678'. In this scenario, I would still like to return the `callid`. Hopefully that makes sense.

**Reply by nblumhardt** [reply]
👍 Sounds like that might just be:

```sql
select callid as id
from stream
where callid is not null and @Message like '%UniqueId: %'
limit 1000
```

Does that help?
